src/cli/javascript/request_filter.rs

```diff
@@ -34,14 +34,25 @@ impl RequestFilter {
     }
 
     #[async_recursion::async_recursion]
-    async fn on_request(&self, mut request: reqwest::Request) -> anyhow::Result<Response<Bytes>> {
+    async fn on_request(
+        &self,
+        mut request: reqwest::Request,
+        http_filter: Option<http::HttpFilter>,
+    ) -> anyhow::Result<Response<Bytes>> {
+        // by default
+        let mut on_request = "onRequest".to_string();
```

Comment:
I think we can drop default. I understand it's a breaking change, but it's less confusing this way.